### PR TITLE
🔥 Remove Hypothesis plugin for now

### LINKED ISSUE #5690: remove currently hypothesis code

### PR DESCRIPTION
**Pull request by lig**
## Change Summary

Remove Hypothesis plugin. Updates docs with a link to the corresponding issue.

## Related issue number

Fix #5690 

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [ ] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @dmontagu

### CODE REVIEW

File: docs/integrations/hypothesis.md [hunk 1]
```diff
@@ -5,53 +5,6 @@ many standard library types, and generic types from the
 [`typing`](https://docs.python.org/3/library/typing.html) and
 [`typing_extensions`](https://pypi.org/project/typing-extensions/) modules by default.
 
-From Pydantic v1.8 and [Hypothesis v5.29.0](https://hypothesis.readthedocs.io/en/latest/changes.html#v5-29-0),
-Hypothesis will automatically load support for [custom types](/usage/types/) like
-`PaymentCardNumber` and `PositiveFloat`, so that the
-[`st.builds()`](https://hypothesis.readthedocs.io/en/latest/data.html#hypothesis.strategies.builds)
-and [`st.from_type()`](https://hypothesis.readthedocs.io/en/latest/data.html#hypothesis.strategies.from_type)
-strategies support them without any user configuration.
+Pydantic v2.0 drops built-in support for Hypothesis and no more ships with the integrated Hypothesis plugin.
```
Comment (samuelcolvin) [inline docs/integrations/hypothesis.md:8]: something about temporarily dropping it while we decide on the way forward, and a link to the issue I just created on annotated types.

File: mkdocs.yml [hunk 2]
```diff
@@ -90,7 +90,6 @@ nav:
 - Integrations:
   - 'Mypy': integrations/mypy.md
   - 'PyCharm': integrations/pycharm.md
-  - 'Hypothesis': integrations/hypothesis.md
```
Comment (samuelcolvin) [inline mkdocs.yml:93]: I think this should stay for clarity.

File: pdm.lock [hunk 3]
```diff
@@ -821,66 +792,66 @@ content_hash = "sha256:9ebdeaedcbb164eedbf015178cc9e587e8721d276099c6077dde1c720
...
-    {url = "https://files.pythonhosted.org/packages/ef/b7/a59cdcb2533488c504af08216f6564cc15c048063a79a15e12e27351d2ca/coverage-7.2.3-cp310-cp310-musllinux_1_1_aarch64.whl", hash = "sha256:4a42e1eff0ca9a7cb7dc9ecda41dfc7cbc17cb1d02117214be0561bd1134772b"},
-    {url = "https://files.pythonhosted.org/packages/ef/b8/fb8b00e74034210dc7969adbd5984b929c4db83ba262a90e87ab2a5cbadc/coverage-7.2.3-cp311-cp311-macosx_10_9_x86_64.whl", hash = "sha256:dfd393094cd82ceb9b40df4c77976015a314b267d498268a076e940fe7be6b79"},
-    {url = "https://files.pythonhosted.org/packages/f8/2b/a5bfe2f7dfb059c3c1f46ff89ac97230121e38f9d970c3ecad78ce1e4aa8/coverage-7.2.3-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl", hash = "sha256:2c2e58e45fe53fab81f85474e5d4d226eeab0f27b45aa062856c89389da2f0d9"},
-    {url = "https://files.pythonhosted.org/packages/fa/19/883184f389f9867ecf14903477c504a65944f55c6520166c67d8799039ba/coverage-7.2.3-cp310-cp310-musllinux_1_1_i686.whl", hash = "sha256:be19931a8dcbe6ab464f3339966856996b12a00f9fe53f346ab3be872d03e257"},
+"coverage 7.2.5" = [
```
Comment (samuelcolvin) [inline pdm.lock:795]: it looks like you deleted and re-created this file. Better to use

```bash
pdm remove -G linting hypothesis
pdm remove -G testing-extra hypothesis
```

**Review by samuelcolvin — COMMENTED** [review]
otherwise LGTM.

File: pdm.lock (hunk 3 above)
Comment (lig) [reply pdm.lock:795]: Just have done `pdm lock`. I'll try the other way.

Comment (lig) [reply pdm.lock]: Does it look better now?

I see how `pdm lock` updated some other pins as well unlike `pdm remove`.

**Comment by lig** [comment]
@samuelcolvin please review

File: docs/integrations/hypothesis.md [hunk 4]
```diff
@@ -5,53 +5,8 @@ many standard library types, and generic types from the
...
 
-!!! warning
-    Please note, while the plugin supports these types, hypothesis will(currently) generate values outside
-    of given args for the constrained function types.
+We are removing the Hypothesis plugin in favor of the mechanism discussed here  removed the plugin as we hope it shouldn’t be necessary in future. We hope to bring back support for Hypothesis to Pydantic via the mechanism discussed in [Way to communicate more information between libraries · Issue #37 · annotated-types/annotated-types](https://github.com/annotated-types/annotated-types/issues/37).
```
Comment (samuelcolvin) [inline docs/integrations/hypothesis.md:10]:
```suggestion
We are temporarily removing the Hypothesis plugin in favor of another mechanism discussed here - [Way to communicate more information between libraries · Issue #37 · annotated-types/annotated-types](https://github.com/annotated-types/annotated-types/issues/37).
```

File: docs/integrations/hypothesis.md [hunk 5]
```diff
@@ -5,53 +5,8 @@ many standard library types, and generic types from the
...
-```
-
-### Use with JSON Schemas
-
-To test client-side code, you can use [`Model.model_json_schema()`](usage/models/) with the
-[`hypothesis-jsonschema` package](https://pypi.org/project/hypothesis-jsonschema/)
-to generate arbitrary JSON instances matching the schema.
-For web API testing, [Schemathesis](https://schemathesis.readthedocs.io) provides
-a higher-level wrapper and can detect both errors and security vulnerabilities.
+It is also possible that Hypothesis plugin will be back as a separate library for Pydantic v2.0+ series (see [V2: hypothesis plugin rewrite · Issue #4682 · pydantic/pydantic](https://github.com/pydantic/pydantic/issues/4682)).
```
Comment (samuelcolvin) [inline docs/integrations/hypothesis.md:12]:
```suggestion
It is also possible that Hypothesis plugin will be back in Pydantic v2.0+ (see [V2: hypothesis plugin rewrite · Issue #4682 · pydantic/pydantic](https://github.com/pydantic/pydantic/issues/4682)).
```

**Review by samuelcolvin — APPROVED** [review]
once that's done, this can be merged.

**Pull request merged**